WordPress/gutenberg
# Parsing patterns when idling (performance follow-up for inserting patterns into containers)

## Description
Related: https://github.com/WordPress/gutenberg/issues/28872 (I don't think this PR fixes the issue. It's an improvement rather than a complete fix)

We need to parse all the patterns to determine whether a pattern can be inserted into the selected destination. We parse them immediately when the editor loads which freezes the editor until the parser finishes. To avoid the thread blocking, this PR changes it to only parse when we have resources to do so.

## How has this been tested?
1. Open block editor
2. Make sure everything works just as before

## Screenshots 

#### Before

Used 1000 dummy patterns for perf testing.

Parsing patterns blocks the editor for ~12 seconds. LCP at ~13500ms
![image](https://user-images.githubusercontent.com/2256104/109637513-2cd84000-7b4d-11eb-90f1-bf0290d86657.png)

#### After
Parsing patterns doesn't block the editor anymore. LCP at ~4200ms
![image](https://user-images.githubusercontent.com/2256104/109637254-d5d26b00-7b4c-11eb-9c55-cb2c9c00290e.png)

## Types of changes
Performance

## Checklist:
